### PR TITLE
Barsign UI/UX Improvements - Emissive effects, balloon alerts, and refactored code

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request
The barsign code is over a decade old so this is a big refactor with some notable improvements:

- Emissive effects (neon lights now glow in the dark)
- Balloon alerts instead of `to_chat` messages
- Mapping helpers based on direction and all_access
- Barsigns are considered machinery now and use power

## Why It's Good For The Game
dreamseeker_6dT8Egk3ow

## Changelog
:cl:
qol: Add emissive effects and balloon alerts to barsigns (neon lights now glow in the dark). Barsigns are now considered machinery and will be affected by power.
refactor: Refactor barsign code to be more robust
/:cl:

### CODE REVIEW

**Comment by ZephyrTFA** [comment]
can you make them give off colored light based on the average of the active sign

File: code/game/objects/structures/barsigns.dm [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/obj/structure/sign/barsign // All Signs are 64 by 32 pixels, they take two tiles
+/obj/machinery/barsign // All Signs are 64 by 32 pixels, they take two tiles
```
Comment (ZephyrTFA) [inline code/game/objects/structures/barsigns.dm:1]: should probably move this out of the structures folder

**Comment by timothymtorres** [comment]
It already does that.

File: code/game/objects/structures/barsigns.dm (hunk 1 above)
Comment (timothymtorres) [reply code/game/objects/structures/barsigns.dm:1]: Good idea!  Applied with https://github.com/tgstation/tgstation/pull/73106/commits/31d48de1f25a15cf9dae6a865a47685a25b413d3

**Comment by ZephyrTFA** [comment]
dont forget to change the dme entry to the new path

**Comment by Profakos** [comment]
Sorry but the update path script is the wrong way around...

**Comment by timothymtorres** [comment]
> Sorry but the update path script is the wrong way around...

Thanks for letting me know.  Fixed in #73268

**Pull request merged**